feg/gateway/services/n7_n40_proxy/metrics/metrics.go

```diff
@@ -0,0 +1,230 @@
+	if err != nil {
+		return nil, err
+	}
+	smPolicyUpdateTotal, err := service_health_metrics.GetInt64("pcf_sm_policy_update_requests_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyUpdateFailures, err := service_health_metrics.GetInt64("pcf_sm_policy_update_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyDeleteTotal, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_requests_total")
+	if err != nil {
+		return nil, err
+	}
+	smPolicyDeleteFailure, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	n7Timeouts, err := service_health_metrics.GetInt64("pcf_sm_policy_delete_failures_total")
+	if err != nil {
+		return nil, err
+	}
+	return &SessionHealthMetrics{
+		SmPolicyCreateTotal:    smPolicyCreateTotal,
+		SmpolicyCreateFailures: smPolicyCreateFailures,
+		SmPolicyUpdateTotal:    smPolicyUpdateTotal,
+		SmPolicyUpdateFailures: smPolicyUpdateFailures,
+		SmPolicyDeleteTotal:    smPolicyDeleteTotal,
+		SmPolicyDeleteFailures: smPolicyDeleteFailure,
+		N7Timeouts:             n7Timeouts,
+	}, nil
+}
+
+func (prevMetrics *SessionHealthMetrics) GetDelta(
+	currentMetrics *SessionHealthMetrics,
+) (*SessionHealthMetrics, error) {
+	if currentMetrics == nil {
+		return nil, fmt.Errorf("GetDelta has nil SessionHealthMetrics on N7N40Proxy")
+	}
+	deltaMetrics := &SessionHealthMetrics{
+		SmPolicyCreateTotal:    currentMetrics.SmPolicyCreateTotal - prevMetrics.SmPolicyCreateTotal,
+		SmpolicyCreateFailures: currentMetrics.SmpolicyCreateFailures - prevMetrics.SmpolicyCreateFailures,
+		SmPolicyUpdateTotal:    currentMetrics.SmPolicyUpdateTotal - prevMetrics.SmPolicyUpdateTotal,
+		SmPolicyUpdateFailures: currentMetrics.SmPolicyUpdateFailures - prevMetrics.SmPolicyUpdateFailures,
+		SmPolicyDeleteTotal:    currentMetrics.SmPolicyDeleteTotal - prevMetrics.SmPolicyDeleteTotal,
+		SmPolicyDeleteFailures: currentMetrics.SmPolicyDeleteFailures - prevMetrics.SmPolicyDeleteFailures,
+		N7Timeouts:             currentMetrics.N7Timeouts - prevMetrics.N7Timeouts,
+	}
+	// Update stored counts to current metric totals
+	*prevMetrics = *currentMetrics
+	return deltaMetrics, nil
+}
+
+// Generic metric functions
+func UpdateN7RecentRequestMetrics(err error) {
+	if err == nil {
+		N7SuccessTimestamp.Set(float64(time.Now().Unix()))
+		N7FailuresSinceLastSuccess.Set(0)
+	} else {
+		N7FailuresSinceLastSuccess.Inc()
+	}
+}
+
+func ReportCreateSmPolicy(err error) {
+	UpdateN7RecentRequestMetrics(err)
+	if err != nil {
+		PcfSmPolicyCreateFailures.Inc()
+	}
+	PcfSmPolicyCreateRequests.Inc()
+}
+
+func ReportUpdateSmPolicy(err error) {
+	UpdateN7RecentRequestMetrics(err)
+	if err != nil {
+		PcfSmPolicyUpdateFailures.Inc()
+	}
+	PcfSmPolicyUpdateRequests.Inc()
+}
+
+func ReportDeleteSmPolicy(err error) {
+	UpdateN7RecentRequestMetrics(err)
+	if err != nil {
+		PcfSmPolicyDeleteFailures.Inc()
+	}
+	PcfSmPolicyDeleteRequests.Inc()
+}
```

Comment:
i know we do not have any test file for any of our health files. Someday we will have to fix that. But lets leave it as it is for now